src/components/profile/UserProfilePage/index.tsx

```diff
@@ -154,16 +154,17 @@ export const UserProfilePage = ({
             )}
           </Typography>
           <Carousel>
-            {(isSignedInUser ? signedInAttendances : (attendances as PublicAttendance[])).map(
-              ({ event }) => (
+            {(isSignedInUser ? signedInAttendances : (attendances as PublicAttendance[]))
+              .reverse()
+              .slice(0, 10)
+              .map(({ event }) => (
```

Comment:
Nit: it's probably more performant to slice the list first then reverse that.
I think that `.slice(-10)` does what we want here